en/phased-ranking.html

```diff
@@ -126,3 +126,180 @@ <h2 id="choosing-phased-ranking-functions">Choosing phased ranking functions</h2
+  This is aggregated per document and makes it easy to analyse using the query and grouping APIs in Vespa -
+  and no other processing/storage is required.
+</p>
+<p>
+  A mutable attribute is a number where an operation can be executed in 4 phases:
+</p>
+<ol>
+  <li>on-match</li>
+  <li>on-first-phase</li>
+  <li>on-second-phase</li>
+  <li>on-summary</li>
+</ol>
+<p>
+  The common use case is to increase the value by 1 for each execution.
+  With this, it is easy to evaluate the document's performance to the queries,
+  e.g. find the documents that appear in most queries, or the ones that never matched -
+  run a query and order by the mutable attribute.
+</p>
+{% include note.html content='The mutable attributes are just counters and memory-operations only -
+the values do not survive content node restarts. The values cannot be compared across nodes.
+Use the values to assess relative document matching and ranking performance since Vespa start' %}
+<p>
+  This example is based on the <a href="vespa-quick-start.html">quickstart</a>.
+  It uses 4 attributes that each track how many times a document participates in any of the 4 phases.
+  This is tracked only if using rank-profile <code>rank_albums_track</code> in the query:
+</p>
+<pre>
+schema music {
+
+    document music {
+
+        field artist type string {
+            indexing: summary | index
+        }
+
+        field album type string {
+            indexing: summary | index
+        }
+
+        field year type int {
+            indexing: summary | attribute
+        }
+
+        field category_scores type tensor&lt;float&gt;(cat{}) {
+            indexing: summary | attribute
+        }
+
+    }
+
+    field match_count type long {
+        indexing: attribute | summary
+        attribute: mutable
+    }
+    field first_phase_count type long {
+        indexing: attribute | summary
+        attribute: mutable
+    }
+    field second_phase_count type long {
+        indexing: attribute | summary
+        attribute: mutable
+    }
+    field summary_count type long {
+        indexing: attribute | summary
+        attribute: mutable
+    }
+
+    fieldset default {
+        fields: artist, album
+    }
+
+    rank-profile rank_albums inherits default {
+        first-phase {
+            expression: sum(query(user_profile) * attribute(category_scores))
+        }
+        second-phase {
+            expression: attribute(year)
+            rerank-count: 1
+        }
+    }
+
+    rank-profile rank_albums_track inherits rank_albums {
+        mutate {
+            on-match        { match_count        += 1 }
+            on-first-phase  { first_phase_count  += 1 }
+            on-second-phase { second_phase_count += 1 }
+            on-summary      { summary_count      += 1 }
+        }
+    }
+
+    rank-profile rank_albums_reset inherits rank_albums {
+        mutate {
+            on-match        { match_count         = 0 }
```

Comment:
To get proper reseting of all 4 mutable attributes, you must define 4 rank profiles where _on-match_ sets the value for that attribute to 0. Only _on-match_ is guaranteed to be executed for all documents when using a 'true' query.

Reply:
Good point! will redo this part and add a ToDo for the rest